connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -128,6 +143,26 @@ private[sql] class AvroDeserializer(
       case (INT, IntegerType) => (updater, ordinal, value) =>
```

Comment:
Shall we clearly define what compatible type is? seems we allow reading interval as int.

Reply:
Maybe we should just fail earlier if the read type is different from the read type.

Reply:
You mean we should fail it as long as the read type is different from the data type?

Reply:
Can you explain more about what you suggest?

Reply:
ok maybe we need to understand the current behavior first. @zeruibao can you read the code and give a summary? Can we read int avro field as long type?

Reply:
For current behavior:
- Interval types can be read as date or timestamp types
- Decimal types can be read with lower precision
- Interval Type can be read as Int or Long.
After fix:
- All above will throw an exception.

For int and long, we always cannot allow this.

Reply:
It seems like we only allow the read type to be the same as the actual type. Can you elaborate why `acceptsType` is better than `DataTypeUtils.sameType`?

Reply:
If we use `acceptsType`, it will cause something like
`Cannot convert Avro type {"type":"record","name":"topLevelRecord","fields":[{"name":"date","type":[{"type":"int","logicalType":"date"},"null"]}]} to SQL type STRUCT<date: DATE>.`
The logical type is the same, but `DataTypeUtils.sameType` will return false, that is not expected. We still allow to read if the logical type is the same.

Reply:
@gengliangwang, could you also take a look at the question?

Reply:
It seems we do want to only allow the read type to be the same as the actual type. Then `DataTypeUtils.sameType` is a good fit here. We need to debug a bit more to see why the date type fails the check.

Reply:
Sorry, I find that I made a stupid mistake. Using `DataTypeUtils.sameType` works well. Just change this. Thanks a lot!